lib/cretonne/meta/semantics/types.py

```diff
@@ -0,0 +1,9 @@
+"""
+The semantics.types module predefines all the Cretone primitive bitvector
+types.
+"""
+from cdsl.types import BVType
+from cdsl.typevar import MAX_BITVEC, int_log2
+
+for width in range(0, int_log2(MAX_BITVEC)+1):
+    BVType(2**width)
```

Comment:
An advantage of using ValueType._registry and lookup by name is that we can use existing code to bind primitive instructions. E.g. in test_elaborate.py I use prim_bvsplit.bv64 etc. I modified BVType.with_bits() to generate types on the fly, and removed semantics/types.py